models/block.go

```diff
@@ -80,6 +121,14 @@ func decodeBlockEvent(event cadence.Event) (*Block, error) {
 		return nil, fmt.Errorf("failed to cadence decode block [%s]: %w", event.String(), err)
 	}
 
+	if event.EventType.ID() == testnetEVMBlockExecutedEventID {
```

Comment:
I would move this to block decoding from database on the model, not on the decoding of events